src/main/kotlin/com/sourcegraph/cody/ui/CodyToolWindowFactory.kt

```diff
@@ -415,19 +435,26 @@ class WebUIProxy(private val host: WebUIHost, private val browser: JBCefBrowserB
   }
 
   private fun handleCefQuery(query: String) {
-    val postMessagePrefix = "{\"what\":\"postMessage\",\"value\":"
-    val setStatePrefix = "{\"what\":\"setState\",\"value\":"
-    when {
-      query.startsWith(postMessagePrefix) -> {
-        val stringEncodedJsonMessage =
-            query.substring(postMessagePrefix.length, query.length - "}".length)
-        host.postMessageWebviewToHost(stringEncodedJsonMessage)
+    val queryObject = JsonParser.parseString(query).asJsonObject
+    val queryWhat = queryObject["what"]?.asString
+
+    when (queryWhat) {
+      "postMessage" -> {
+        val queryValue = queryObject["value"] ?: return
+        host.postMessageWebviewToHost(queryValue.toString())
+
+        val messageObject = if (queryValue.isJsonObject) queryValue.asJsonObject else return
+        if (messageObject["command"]?.asString == "copy" &&
```

Comment:
works for now but eventually - if we want to support more different messages - it would be nice to have the handlers in separate methods and keep it cleaner